pkg/controller/zerocapacity/zero_controller.go

```diff
@@ -0,0 +1,465 @@
+			return true
+		}
+	}
+	return false
+}
+
+func (z *Controller) configureBackupPod(name string, configMap *corev1.ConfigMap, ispn *v1.Infinispan, zeroSpec *Spec,
+	pod *corev1.Pod, podSpec corev1.PodSpec) {
+
+	// Utilise the provided PodSpec as a base, overriding fields as required
+	pod.Spec = podSpec
+	var identitiesVol corev1.Volume
+	for _, v := range podSpec.Volumes {
+		if v.Name == ispnCtrl.IdentitiesVolumeName {
+			identitiesVol = v
+		}
+	}
+
+	dataVolName := name + "-data"
+	container := &pod.Spec.Containers[0]
+	container.Name = name
+	container.Resources = zeroSpec.Container.AsResourceRequirements()
+	container.VolumeMounts = []corev1.VolumeMount{
+		{
+			Name:      ispnCtrl.ConfigVolumeName,
+			MountPath: consts.ServerConfigRoot,
+		},
+		{
+			Name:      ispnCtrl.IdentitiesVolumeName,
+			MountPath: consts.ServerSecurityRoot,
+		},
+		// Override the statefulset data volume with Ephemeral vol as we're only interested in data related to CR
+		{
+			Name:      dataVolName,
+			MountPath: ispnCtrl.DataMountPath,
+		},
+		// Mount configured volume at /zero path so that any created content is stored independent of server data
+		{
+			Name:      name,
+			MountPath: zeroSpec.Volume.MountPath,
+		},
+	}
+
+	pod.Spec.SecurityContext.FSGroup = pointer.Int64Ptr(185)
+	pod.Spec.Volumes = []corev1.Volume{
+		// Volume for mounting zero-capacity yaml configmap
+		{
+			Name: ispnCtrl.ConfigVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{Name: configMap.Name},
+				},
+			}},
+		// Volume for reading/writing data
+		{
+			Name:         name,
+			VolumeSource: zeroSpec.Volume.VolumeSource,
+		},
+		// EmptyDir for Infinispan data volume as the zero-capacity node does not store traditional data
+		{
+			Name: dataVolName,
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+		// Volume for identities file
+		identitiesVol,
+	}
+
+	ispnCtrl.AddVolumeForEncryption(ispn, &pod.Spec)
+}
+
+func (z *Controller) configureZeroCapacity(name, namespace string, spec *Spec, infinispan *v1.Infinispan, instance Resource) (*corev1.ConfigMap, error) {
+	clusterConfig := &corev1.ConfigMap{}
+	clusterConfigName := ispnCtrl.ServerConfigMapName(instance.Cluster())
+	clusterKey := types.NamespacedName{
+		Namespace: namespace,
+		Name:      clusterConfigName,
+	}
+	ctx := context.Background()
+	if err := z.Client.Get(ctx, clusterKey, clusterConfig); err != nil {
+		if errors.IsNotFound(err) {
+			return nil, fmt.Errorf("Unable to load ConfigMap: %s", clusterConfigName)
+		}
+		return nil, err
+	}
+
+	yaml := clusterConfig.Data[consts.ServerConfigFilename]
+	config, err := configuration.FromYaml(yaml)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to parse existing config: %s", clusterConfigName)
+	}
+	config.Infinispan.ZeroCapacityNode = true
```

Comment:
Oh, @ryanemerson you change the configuration on the basis of which other Pods working. What happens if `kubelet` decide to reschedule other containers in the Pod at the cluster, they will affect ZeroCapacityNode at all. Not sure that's correct behavior.